devito/operations/interpolators.py

```diff
@@ -388,3 +389,520 @@ def callback():
+                                                 condition=condition, indirect=True)
+            # Track Indexed substitutions
+            idx_subs.append(mapper)
+
+        # Temporaries for the position
+        temps = [Eq(v, k, implicit_dims=self.sfunction.dimensions)
+                 for k, v in self.sfunction._position_map.items()]
+
+        temps.extend([Eq(v, k.subs(self.sfunction._position_map),
+                      implicit_dims=self.sfunction.dimensions)
+                      for k, v in self.sfunction._relative_position_map.items()])
+
+        # Temporaries for the indirection dimensions
+        temps.extend([Eq(v, k.subs(self.sfunction._position_map),
+                         implicit_dims=self.sfunction.dimensions)
+                      for k, v in points.items()])
+
+        return idx_subs, temps
+
+    def interpolate(self, expr, offset=0, increment=False, self_subs={}):
+        """
+        Generate equations interpolating an arbitrary expression into ``self``.
+        Parameters
+        ----------
+        expr : expr-like
+            Input expression to interpolate.
+        offset : int, optional
+            Additional offset from the boundary.
+        increment: bool, optional
+            If True, generate increments (Inc) rather than assignments (Eq).
+        """
+        def callback():
+
+            # Derivatives must be evaluated before the introduction of indirect accesses
+            try:
+                _expr = expr.evaluate
+            except AttributeError:
+                # E.g., a generic SymPy expression or a number
+                _expr = expr
+
+            variables = list(retrieve_function_carriers(_expr))
+
+            # Need to get origin of the field in case it is staggered
+            # TODO: handle each variable staggereing spearately
+            field_offset = variables[0].origin
+
+            idx_subs, temps = self._interpolation_indices(variables, offset,
+                                                          field_offset=field_offset)
+
+            rhs = Symbol(name='sum', dtype=self.sfunction.dtype)
+            summands = [Eq(rhs, 0., implicit_dims=self.sfunction.dimensions)]
+
+            # Verify if is a 2D or 3D interpolation
+            dim_pos = self.sfunction.grid.dimensions
+            if len(dim_pos) == 2:
+                result = self._sinc_equations2D(_expr, idx_subs, dim_pos=dim_pos)
+            else:
+                result = self._sinc_equations3D(_expr, idx_subs, dim_pos=dim_pos)
+
+            eqs = [np.sum(v) for v in result]
+
+            summands.extend([Inc(rhs, v, implicit_dims=self.sfunction.dimensions)
+                             for v in eqs])
+
+            # Write/Incr `self`
+            lhs = self.sfunction.subs(self_subs)
+            last = [Inc(lhs, rhs)] if increment else [Eq(lhs, rhs)]
+
+            # Creates the symbolic equation that calls the populate function
```

Comment:
What's this?

Reply:
To implement the sinc interpolation I used a look up table that holds the values of the coefficients. Before it could be accessed, it was necessary to initialize this table with the values. This symbolic Function `populate()` creates a call to the function responsible for initialize the look up table.

This function is defined inside look up table.h, that I apparently forgot to add here. I'm attaching the lookuptable.h in this comment, this way you can see how it works

[lookuptable.zip](https://github.com/devitocodes/devito/files/8872262/lookuptable.zip)